Refactored "open from toasts"-implementation

TranRed opened the pull request:
I think I kinda combined suggestions 2&3 from last merge:

2. Keep two maps (one map<enum class, string>, one map<string, enum class>) in parallel 
3. Make two functions, one for converting an enum class to string with a  simple switch-case, one for converting a string to an enum class with  some if/elseif stuff

Please review and maybe squash commits (?)

![grafik](https://user-images.githubusercontent.com/16636423/56849347-95bc5480-68f3-11e9-8024-ff9b4c983cc0.png)

pajlada commented on src/singletons/Toasts.hpp:
```diff
@@ -7,17 +7,29 @@ namespace chatterino {
 
 enum class Platform : uint8_t;
 
+enum class ToastReactions {
```
The casing(?) of the enums don't match the rest of the codebase, we use PascalCase, not camelCase
see https://github.com/fourtf/chatterino2/blob/3f41dfeff942d9becb311a4b1769e8445da016af/src/messages/layouts/MessageLayout.hpp#L24-L29

TranRed commented:
Added a new commit. Changing to Int Setting "broke" the ComboBox I used. I implemented my own combobox for now, not really sure about that implementations. Doesn't feel high quality to me, but works for now. Should prolly be changed in later [refactor](https://geekandpoke.typepad.com/geekandpoke/2010/11/todoxls.html) . I assume it needs to get reworked anyways once we will try to get rid of the mix of "new" (e.g. general, external tools) and "old" (e.g. commands, highlights, notifications) setting pages.

TranRed commented on src/widgets/settingspages/NotificationPage.cpp:
```diff
@@ -133,4 +134,31 @@ NotificationPage::NotificationPage()
+        },
+        managedConnections);
+
+    QObject::connect(toastReactionOptions, &QComboBox::currentTextChanged,
```
Couldn't figure out how to properly connect the currentIndexChanged Signal (with int), to update the setting from that.

TranRed replied:
Now the proper event is connected thanks to @pajlada FeelsGoodManClap

TranRed commented on src/singletons/Toasts.cpp:
```diff
@@ -35,6 +48,38 @@ bool Toasts::isEnabled()
+    return Toasts::findStringFromReaction(static_cast<ToastReactions>(i));
+}
+
+ToastReactions Toasts::findReactionFromString(const QString &string)
```
I just realised this is probably not used anymore, because we connected the proper signal to the combobox now. I want to delete the unused before merging. Will check and commit that later!

pajlada commented on src/singletons/Toasts.hpp:
```diff
@@ -7,17 +7,29 @@ namespace chatterino {
 
 enum class Platform : uint8_t;
 
+enum class ToastReactions {
```
Should be called `ToastReaction` (singular) not `ToastReactions` (plural)

TranRed replied:
done
